src/security/security-tests/src/test/java/org/geoserver/security/impl/SecureCatalogImplTest.java

```diff
@@ -1151,8 +1151,8 @@ public void testSecurityFilterComplex() throws Exception {
         Collection<PublishedInfo> filteredPublished =
                 Collections2.filter(publisheds, new PredicateFilter(security));
         assertEquals(4, filteredLayers.size());
-        assertEquals(0, filteredGroups.size());
-        assertEquals(4, filteredPublished.size());
+        assertEquals(2, filteredGroups.size());
+        assertEquals(6, filteredPublished.size());
```

Comment:
These numbers agree with your fix; but not with the test case comment above:
```
        // Check the count does not include the groups. Since the catalog is just a mock, we
        // extract the layers and groups, and do counts using the security filter
```

So the comment is "accurate" documenting the incorrect behaviour that you have now fixed :)

I have *no idea* why the previous behaviour of the security filter was to ignore layer groups; it makes no sense as we need those to be included for layer preview page to include them?!?

Reply:
I think it was probably missed because this functionality is not turned on if you are admin or not filtering by workspace.  I think your changes to the homepage made this more visible and used.  Good to catch it!

Reply:
I agree; renaming the variables to improve code readability, and stepping through the code multiple times with the debugger has me confident the fix is now doing the right thing.